packages/jellyfish-api-core/__tests__/category/icxorderbook/claimDFCHTLC.test.ts

```diff
@@ -0,0 +1,233 @@
+    const listHTLCOptions: ICXListHTLCOptions = {
+      offerTx: makeOfferTxId,
+      closed: true
+    }
+    const HTLCs: Record<string, ICXDFCHTLCInfo| ICXEXTHTLCInfo| ICXClaimDFCHTLCInfo> = await container.call('icx_listhtlcs', [listHTLCOptions])
+    expect(Object.keys(HTLCs).length).toBe(4) // extra entry for the warning text returned by the RPC atm.
+    // we have a common field "type", use that to narrow down the record
+    if (HTLCs[claimTxId].type === ICXHTLCType.CLAIM_DFC) {
+      // ICXClaimDFCHTLCInfo cast
+      const ClaimHTLCInfo: ICXClaimDFCHTLCInfo = HTLCs[claimTxId] as ICXClaimDFCHTLCInfo
+      expect(ClaimHTLCInfo.dfchtlcTx).toStrictEqual(DFCHTLCTxId)
+      expect(ClaimHTLCInfo.seed).toStrictEqual('f75a61ad8f7a6e0ab701d5be1f5d4523a9b534571e4e92e0c4610c6a6784ccef')
+    }
+
+    // check HTLC DFCHTLCTxId is in claimed status
+    if (HTLCs[DFCHTLCTxId].type === ICXHTLCType.DFC) {
+      // ICXDFCHTLCInfo cast
+      const DFCHTLCInfo: ICXDFCHTLCInfo = HTLCs[DFCHTLCTxId] as ICXDFCHTLCInfo
+      expect(DFCHTLCInfo.status).toStrictEqual(ICXHTLCStatus.CLAIMED)
+    }
+
+    const accountDFIAfterClaim = await container.call('getaccount', [accountDFI, {}, true])
+    const accountBTCAfterClaim = await container.call('getaccount', [accountBTC, {}, true])
+
+    // maker should get incentive + maker deposit and taker should get amount in DFCHTLCTxId HTLC - takerfee
+    // expect(Number(accountDFIAfterClaim[idDFI])).toStrictEqual(Number(accountDFIBeforeClaim[idDFI]) + Number(0.01500000000000) + Number(0.00375000000000))
+    expect(Number(accountDFIAfterClaim[idDFI]).toPrecision(8)).toStrictEqual((Number(accountDFIBeforeClaim[idDFI]) + Number(0.01500000000000) + Number(0.00375000000000)).toPrecision(8))
+    expect(Number(accountBTCAfterClaim[idDFI])).toStrictEqual(Number(accountBTCBeforeClaim[idDFI]) + Number(15))
+  })
+
+  it('Should claim DFCHTLC with input UTXOs', async () => {
+    const { makeOfferTxId, DFCHTLCTxId } = await setupUntilSubmitExtHTLCForDFCBuyOffer()
+
+    const accountDFIBeforeClaim = await container.call('getaccount', [accountDFI, {}, true])
+    const accountBTCBeforeClaim = await container.call('getaccount', [accountBTC, {}, true])
+
+    // input utxos
+    const utxos = await container.call('listunspent', [1, 9999999, [accountDFI], true])
+    const inputUTXOs: InputUTXO[] = utxos.map((utxo: InputUTXO) => {
+      return {
+        txid: utxo.txid,
+        vout: utxo.vout
+      }
+    })
+    // claim
+    const claimTxId = (await client.icxorderbook.claimDFCHTLC(DFCHTLCTxId, 'f75a61ad8f7a6e0ab701d5be1f5d4523a9b534571e4e92e0c4610c6a6784ccef', inputUTXOs)).txid
+    await container.generate(1)
+
+    // List htlc and check
+    const listHTLCOptions: ICXListHTLCOptions = {
+      offerTx: makeOfferTxId,
+      closed: true
+    }
+    const HTLCs: Record<string, ICXDFCHTLCInfo| ICXEXTHTLCInfo| ICXClaimDFCHTLCInfo> = await container.call('icx_listhtlcs', [listHTLCOptions])
```

Comment:
Here the client.icxorderbook.listHTLCs() function can return 3 types of structures depending on the input. I used a union for this. But on the client side code, returned structure has to be narrowed down depends on either the input or certain properties of the output structure. Here it is narrowed down based on HTLCs[claimTxId].type.
If anybody has better idea to deal this kind of situations please comment.